### PR TITLE
feat: Add octokit and fetching users/repos for workspaces flows

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

This adds octokit (alongside a personal web token auth method) to fetch github users / repos when workspaces flows add unknown users/repos.

The biggest thing is that we need the user/repo ID: because we don't generate IDs as the primary key for either the user table nor the repo table, we get those directly from GitHub. So attempting to come up with a "temporary" or partial entry that can latter get picked up by the ETL is abit of a mess.

Instead, this uses the octokit directly and gets those entities and fills in the necessary partial entries to be filled in at a latter time by the ETL.

---

This also refactors the workspaces repos and contributors service functions to use a common "execute..." function that performs that functionality.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes: https://github.com/open-sauced/api/issues/511

## Steps to QA

TODO - still doing some deeper validation with the ETL too.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by jpmcb** [comment]
cc @open-sauced/engineering  - opening this now to get some validation this is the right approach: we'll need another github auth token or spend the time getting this integrated with a github org app.

**Comment by jpmcb** [comment]
Force pushed to add some additional checks on the repo stub creation: now, users (which get referenced in the foreign key `user_id`) get created correctly as well.

File: src/insight/dtos/create-insight.dto.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class CreateInsightDto {
     description: "An array of repository information objects",
     isArray: true,
     type: RepoInfo,
-    example: [{ id: 797, full_name: "open-sauced/insights" }],
+    example: [{ id: 797, fullName: "open-sauced/insights" }],
```
Comment (nickytonline) [inline src/insight/dtos/create-insight.dto.ts:28]: is this change because we're bringnig in Octokit, so JS camelCase?

Comment (jpmcb) [reply src/insight/dtos/create-insight.dto.ts:28]: This was a typo in the swagger stuff: the field for that DTO isn't `full_name`, it's `fullName` which was erroring out when trying to use the swagger UI. So this is just a one off typo fix 😅

Comment (nickytonline) [reply src/insight/dtos/create-insight.dto.ts:28]: 😅

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**